specification/sbi_atee.adoc

```diff
@@ -13,6 +44,936 @@ running a VMM).
+
+
+=== VCPU assignment
+
+The host must register platform VCPUs with the TSM before they can be used for TVM execution by calling
+`sbi_tee_tvm_vcpu_create()`.
+
+=== TVM execution
+
+Following assignment of memory and VCPU resources, the host can transition the guest into a "runnable"
+state by calling `sbi_tee_tvm_finalize()`. Note that some TEE calls are no longer permissble after this
+transition.
+
+The host can use the `sbi_tee_tvm_vcpu_set_register()` to setup TVM execution parameters like the entry
+point, and then call `sbi_tee_tvm_vcpu_run()` to begin execution. TVM execution continues until there an
+event like an interrupt, or fault that cannot be serviced by the TSM. The host can examine the return code
+from the ECALL, and use `sbi_tee_tvm_get_vcpu_register()` to determine further course of action. This may
+involve serving page-faults for demand-page regions, servicing MMIO requests, or terminating the TVM in
+case of an irrecoverable exception.
+
+Page faults and MMIO requests can be processed using `sbi_tee_tvm_add_zero_pages()`, `sbi_tee_tvm_add_shared_pages()`,
+and `sbi_tee_tvm_vcpu_set_register()`, and then calling `sbi_tee_tvm_vcpu_run()` to resume execution.
+
+=== TVM teardown
+
+The host can teardown a TVM by calling `sbi_tee_destroy_tvm()`. This automatically releases all confidential memory
+assigned to the TVM, and it can be repurposed for use with other TVMs. However, reclaiming the memory for use by
+non-confidential workloads requires an explicit call to `sbi_tee_tsm_reclaim_pages()`.
+
+== TEE Extension (EID #0x41544545)
+
+=== Listing of common enums
+The following enums are referenced by several functions described below.
+
+[source, C]
+-------------------
+enum tsm_page_type {
+    /* 4KiB */
+    PAGE_4K = 0,
+    /* 2 MiB */
+    PAGE_2MB = 1,
+    /* 1 GiB */
+    PAGE_1GB = 2,
+    /* 512 GiB */
+    PAGE_512GB = 3,
+}
+-------------------
+
+[source, C]
+-----
+enum tvm_vcpu_register {
+    /*
+     * Entry point (initial SEPC) of the boot CPU of a TVM. Read-write prior to TVM
+     * finalization; inaccessible after the TVM has started.
+     */
+    ENTRY_PC = 0,
+    /*
+     * Boot argument (stored in A1, usually a pointer to a device-tree) of the boot CPU
+     * of a TVM. Read-write prior to TVM finalization; inaccessible after the TVM has
+     * started.
+    */
+    ENTRY_ARG = 1,
+    /*
+     * Detailed TVM CPU exit cause register. Read-only, and only accessible after the
+     * TVM has started.
+     */
+    EXIT_CAUSE_0 = 2,
+    /* An additional exit cause register with the same access properties as exit_cause_0. */
+    EXIT_CAUSE_1 = 3,
+    /*
+     * Value used to complete an emulated MMIO load by a TVM CPU. Read-write, and only
+     * accessible after the TVM has started.
+     */
+    MMIO_LOAD_VALUE = 4,
+    /*
+     * Value stored by a TVM CPU to an emulated MMIO region. Read-only, and only accessible
+     * after the TVM has started.
+     */
+    MMIO_STORE_VALUE = 5,
+};
```

Comment:
so this structure is going to change right?

Reply:
Right...will wait for PR to be finalized.

Reply:
I think this can be removed now?